source/rules/just.ts

```diff
@@ -0,0 +1,61 @@
+/**
+ * @license Use of this source code is governed by an MIT-style license that
+ * can be found in the LICENSE file at https://github.com/cartant/eslint-plugin-rxjs
+ */
+
+import { Rule } from "eslint";
+import esquery from "esquery";
+import * as es from "estree";
+import { getParent, typecheck } from "../utils";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      category: "RxJS",
+      description: "Enforces the use of a `just` alias for `of`.",
+      recommended: false
+    },
+    fixable: null,
+    messages: {
+      forbidden: "Use just alias."
+    },
+    schema: []
+  },
+  create: context => {
+    const { couldBeObservable } = typecheck(context);
+
+    return {
+      "ImportDeclaration[source.value='rxjs'] > ImportSpecifier[imported.name='of']": (
+        node: es.ImportSpecifier
+      ) => {
+        const { loc } = node;
+        context.report({
+          messageId: "forbidden",
+          loc: {
+            ...loc,
+            end: {
+              ...loc.start,
+              column: loc.start.column + 2
+            }
+          }
+        });
+
+        const ofIdentifiers = esquery(
+          context.getSourceCode().ast,
+          "CallExpression > Identifier[name='of']"
+        ) as es.Identifier[];
+        ofIdentifiers.forEach(ofIdentifier => {
+          const callExpression = getParent(ofIdentifier);
+          if (couldBeObservable(callExpression)) {
```

Comment:
Is it OK to verify `of` like this? 
Or do we want to verify that this node is the imported node?

Reply:
Maybe okay for verification, but it won't be okay for fixing, as fixing must not break code.

The TSLint version uses type information to determine that the import actually is the declaration: https://github.com/cartant/rxjs-tslint-rules/blob/a164ffc968521925328cab995a2a8a959e5b4fd4/source/rules/rxjsJustRule.ts#L78

An alternative is to use something that shows you any overrides that might have occurred due to scope. The TSLint rules use this:

https://github.com/cartant/rxjs-tslint-rules/blob/master/source/support/scope-walker.ts

I think some of the remaining rules are going to need something that helps you track scope overrides. I was anticipating writing some ESLint helper for this, but if you are keen...

Reply:
Got it 👍 that's what I thought and that's why I've put a comment.
I'll leave this as is, and come back to it later on.

Reply:
I didn't find a way to implement the `getDeclarations` from tsutils without using typescript. But I found out that we can access the `references` of a variable by looking at the `no-unused-vars` [source code](https://github.com/eslint/eslint/blob/master/lib/rules/no-unused-vars.js), what do you think?

Reply:
Hmm, do we really need to track references here, though. I've only looked at this quickly, but `defs` looks pretty much like what we'd want:

https://github.com/eslint/eslint/blob/e10eebab4abd193dee697c4de7fb2d95bbab2d8c/lib/rules/no-unused-vars.js#L155-L169

If the definition points to the node in the import statement, then it's fixable, right?

Reply:
To be honest, I have no idea.
I've tried it but I'm not coming to a working solution, my trials only led me to the function `context.getDeclaredVariables(node)`.

Reply:
I'll have a look at it soon. Fixing/refactoring `no-finnish`, ATM. Almost done.